### DISCUSSION TITLE
Synology DS220 DSM 7

### DISCUSSION THREAD

**Discussion by RyanGuyCode** [haugene/docker-transmission-openvpn#2205] [Q&A] [answered]
I have the below configuration, and everything starts just fine, and I'm able to download... however, I have 1 volume and for some reason, the way I have set up my system, the "complete" path still ends up inside the /data/config directory on the docker, rather than on one of the separate shared paths as configured. The only way I see the downloads is by connecting to the container and finding it in the filesystem there.

I'm not sure I'm understanding how to change the path effectively. I am on the "latest".  This is how I built the stack.
Do I need to add my download/incomplete path differently to override the settings.json?

version: '3.3'
services:
    transmission-openvpn:
        cap_add:
            - NET_ADMIN
        volumes:
            - '/volume1/docker/transmission/data:/config'
            - '/volume1/data/transmission/downloads:/downloads'
            - '/volume1/data/transmission/incomplete:/incomplete'
        environment:
            - OPENVPN_PROVIDER=NORDVPN
            - CREATE_TUN_DEVICE=true
            - NORDVPN_COUNTRY=US
            - OPENVPN_USERNAME=username
            - OPENVPN_PASSWORD=password
            - LOCAL_NETWORK=192.168.0.0/16,10.26.0.0/24
            - TZ=America/Chicago
            - OPENVPN_OPTS=--inactive 3600 --ping 10 --ping-exit 60
            - restart=always
            - dns=8.8.8.8
            - dns=4.4.4.4
        logging:
            driver: json-file
            options:
                max-size: 10m
        ports:
            - '9091:9091'
        image: haugene/transmission-openvpn

**Comment by pkishino** [comment, marked as answer]
Yes, you need to set the transmission values for the folders, see here:
http://haugene.github.io/docker-transmission-openvpn/config-options/
Plus, you haven’t set your :/data mount as per default..

**Reply by RyanGuyCode** [reply]
ok, I need to check that, because I'm not 100% sure I understand. Do you have an example to follow?

**Reply by RyanGuyCode** [reply]
I'm definitely not getting something here... I thought I set the variables for the directories on my docker:

   volumes:
            - /volume1/docker/transmission/data:/config
            - /volume1/data/transmission/downloads:/downloads
            - /volume1/data/transmission/incomplete:/incomplete
            - /volume1/data/transmission/watch:/watch
        environment:
            - OPENVPN_PROVIDER=NORDVPN
            - CREATE_TUN_DEVICE=true
            - NORDVPN_COUNTRY=US
            - OPENVPN_USERNAME=UN
            - OPENVPN_PASSWORD=PW
            - LOCAL_NETWORK=192.168.0.0/16,10.26.0.0/24
            - TZ=America/Chicago
            - OPENVPN_OPTS=--inactive 3600 --ping 10 --ping-exit 60
            - TRANSMISSION_HOME=/volume1/docker/transmission/data/config/transmission-home
            - TRANSMISSION_DOWNLOAD_DIR=/volume1/data/transmission/downloads
            - TRANSMISSION_INCOMPLETE_DIR=/volume1/data/transmission/incomplete
            - TRANSMISSION_WATCH_DIR=/volume1/data/transmission/watch
            - restart=always
However, the paths still end up in my docker image, not on the NAS, as do any DLs

**Reply by pkishino** [reply]
These are set incorrect:
``
- TRANSMISSION_HOME=/volume1/docker/transmission/data/config/transmission-home
- TRANSMISSION_DOWNLOAD_DIR=/volume1/data/transmission/downloads
- TRANSMISSION_INCOMPLETE_DIR=/volume1/data/transmission/incomplete
- TRANSMISSION_WATCH_DIR=/volume1/data/transmission/watch

``
The transmission variables are internal so you need to set them as per container path..
As you are just using default for the folders, try just specifying 
TRANSMISSION_HOME=/config

**Reply by RyanGuyCode** [reply]
ok, so I have a /docker/ directory --> TRANSMISSION_HOME (for config, right?)
Then the DOWNLOAD,INCOMPLETE,WATCH -->/data/transmission/ (separate partition for storage).

That is what I thought... maybe I don't understand the significance of TRANSMISSION_HOME.

**Reply by pkishino** [reply]
I think you are confusing some docker concepts with the app configs here..
the volume mounts can be whatever you want..they are there to expose your local file system to the docker container. This way, if you expose a folder there you can access it outside AND inside the container, the mapping specifies where to mount the folder inside the container..
anyway, for your case you are overcomplicating things.
all you need is:
```
volumes:
- /volume1/docker/transmission/data:/config
- /volume1/data/transmission/downloads:/data
```
then remove the others as you would be using defaults anyway..but if you want to set them just set like this:
```
- TRANSMISSION_HOME=/config/transmission-home
- TRANSMISSION_DOWNLOAD_DIR=/data/downloads
- TRANSMISSION_INCOMPLETE_DIR=/data/incomplete
- TRANSMISSION_WATCH_DIR=/data/watch
```

for transmission config variables..https://github.com/transmission/transmission/blob/main/docs/Environment-Variables.md

**Reply by RyanGuyCode** [reply]
i see. yes, I think it was a conceptual problem. Thanks. I'll review and test

**Reply by RyanGuyCode** [reply]
@pkishino Ok, so I did make those changes, it seems unstable, at this point, but the logs aren't giving me enough information to work with. Is this a case, that I need to play with those env variables?

version: '3.3'
services:
    transmission-openvpn:
        cap_add:
            - NET_ADMIN
        volumes:
            - /volume1/docker/transmission/data:/config
            - /volume1/data/transmission/downloads:/data
        environment:
            - OPENVPN_PROVIDER=NORDVPN
            - CREATE_TUN_DEVICE=true
            - NORDVPN_COUNTRY=US
            - OPENVPN_USERNAME=
            - OPENVPN_PASSWORD=
            - LOCAL_NETWORK=192.168.0.0/16,10.26.0.0/24
            - TZ=America/Chicago
            - OPENVPN_OPTS=--inactive 3600 --ping 10 --ping-exit 60
            - TRANSMISSION_HOME=/config/transmission-home
            - TRANSMISSION_DOWNLOAD_DIR=/data/downloads
            - TRANSMISSION_INCOMPLETE_DIR=/data/incomplete
            - restart=always
            - dns=8.8.8.8
            - dns=4.4.4.4
            - PUID=1026
            - PGID=100
        logging:
            driver: json-file
            options:
                max-size: 10m
        restart: always
        ports:
            - '9091:9091'
        image: haugene/transmission-openvpn

**Reply by pkishino** [reply]
Can you check if you have enabled the following setting for transmission? Check in the settings file or in the UI:
start-added-torrents
default is True but might have set it to false

**Reply by RyanGuyCode** [reply]
@pkishino Thanks! I think part of the issue was share permissions! Thanks for the hand! I'm smarter by 10% after this!
